src/Corax/Queries/Meta/MergeHelper.cs

```diff
@@ -87,6 +87,9 @@ internal static unsafe int AndVectorized(long* dst, int dstLength, long* left, i
                         continue;
                     }
 
+                    if (largerEndPtr - largerPtr < N)
```

Comment:
We have similar code in Sort Helper, no?

Reply:
SortHelper is not vectorized.